### ISSUE TITLE
Refactoring: analyze and redesign HtmlImage and its usages 

### ISSUE THREAD

**Issue by eddiemuc** [cgeo/cgeo#11961]
Meanwhile I took a deeper look into the HtmlImage class which is in the heart of all this. Frankly I am a bit shocked...
* There seems to be two completely separate image getting and processing/caching mechanism to be implemented there in parallel, one used via getDrawable() method and the other via fetchDrawable. Among other things, they use two different caches
* Caching is implemented on an instance level only. But in many many places in c:geo, HtmlImages are created freshly as needed - in this case the whole caching algorithm becomes useless of course

I see the following caching-anti-patterns througout in the code:

```
        final HtmlImage imgGetter = new HtmlImage(HtmlImage.SHARED, false, false, false);
        imgGetter.fetchDrawable(url)).subscribe(image -> { ... do something })
```

Or, in the blocking variant:

```
        final HtmlImage imgGetter = new HtmlImage(HtmlImage.SHARED, false, false, false);
        final drawable image = imgGetter.getDrawable(myUrl))
```

Implementors have to be aware that these usage patterns **do not use caching** because the `HtmlImage `instance is created freshly for each request!

A usage pattern where caching is really done is e.g. implemented in class `cgeo.geocaching.ui.DirectionImage` -> here a single HtmlImage instance is maintained and caching will actually work.

I will now implement a central `AvatarImage` class (I take DirectionImage as an example) to implement **real** caching for avatar images. 

However, IMHO there's a general rework needed around usage and design of HtmlImage. I think c:geo currently re-downloads **for** more images than needed from the web again and again. But this would be its own issue.

_Originally posted by @eddiemuc in https://github.com/cgeo/cgeo/issues/11227#issuecomment-950124868_
